Allow customizing 'On'/'Off' text in ToggleSwitch

cheshire137 opened the pull request:
This makes the 'On' and 'Off' text that appears next to a ToggleSwitch be customizable. The prop names I chose feel awkward, but I also thought it'd be confusing to have a name like 'onLabel' or 'onText' since it reads as if it's meant to be a function to act as a handler.

One thing I wondered about: was there any accessibility reason why the text was hard-coded to 'On' and 'Off'?

<img width="701" height="444" alt="screenshot of Storybook of ToggleSwitch component in off state with the word Inactive next to the switch" src="https://github.com/user-attachments/assets/4c67dea7-db63-4e6d-8a8a-686ab3f71a84" />

### Changelog

<!-- Under the headings below, list out relevant API changes that this Pull Request introduces -->

#### New

Adds the `buttonLabelOn` and `buttonLabelOff` props to ToggleSwitch to allow changing the 'On' and 'Off' text that appears next to the switch.

#### Changed

<!-- List of things changed in this PR -->

#### Removed

<!-- List of things removed in this PR -->

### Rollout strategy

<!-- How do you recommend this change to be rolled out? Refer to [contributor docs on Versioning](https://github.com/primer/react/blob/main/contributor-docs/versioning.md) for details. -->

- [ ] Patch release
- [x] Minor release
- [ ] Major release; if selected, include a written rollout or migration plan
- [ ] None; if selected, include a brief description as to why

### Testing & Reviewing

<!-- Describe any specific details to help reviewers test or review this Pull Request -->

### Merge checklist

- [x] Added/updated tests
- [x] Added/updated documentation
- [x] Added/updated previews (Storybook)
- [x] Changes are [SSR compatible](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#ssr-compatibility)
- [x] Tested in Chrome
- [ ] Tested in Firefox
- [ ] Tested in Safari
- [ ] Tested in Edge
- [ ] (GitHub staff only) Integration tests pass at github/github ([Learn more about how to run integration tests](https://github.com/github/primer-engineering/blob/main/how-we-work/testing-primer-react-pr-at-dotcom.md))

<!-- Take a look at the [What we look for in reviews](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#what-we-look-for-in-reviews) section of the contributing guidelines for more information on how we review PRs. -->

cheshire137 commented on package-lock.json:
This change was generated when I ran `npx @changesets/cli`.

TylerJDev commented on packages/react/src/ToggleSwitch/ToggleSwitch.docs.json:
```diff
@@ -112,7 +115,19 @@
+      "name": "buttonLabelOn",
+      "type": "string",
+      "defaultValue": "'On'",
+      "description": "The text to display when the toggle switch is turned on."
```
Could we add a note that "on/off" labels should be used if there's better context than the default? 

Quote from Fluent's design system: 

> Only replace the On/Off labels if there are more specific labels for the setting. For example, you might use Show/Hide if the setting is “Show images”.

Maybe we could add something similar? (e.g. The "on/off" labels should only be replaced if the adjusted label provides enhanced context over the default). 

Let me know what you think 😁

TylerJDev commented on packages/react/src/ToggleSwitch/ToggleSwitch.tsx:
```diff
@@ -42,6 +42,10 @@ export interface ToggleSwitchProps extends Omit<React.HTMLAttributes<HTMLDivElem
   loadingLabel?: string
   /** type of button to account for behavior when added to a form*/
   buttonType?: 'button' | 'submit' | 'reset'
+  /** Text to display when the toggle switch is in the 'on' position. Defaults to 'On'. */
```
Just a copy from my last comment, could we add a note stating that labels should only be changed if it enhances the context over the default?

TylerJDev reviewed (approved):
LGTM! Left some comments, but happy to add these in a separate PR too 😁

TylerJDev commented:
For the CI, I'm thinking we just need to add the `update snapshots` label, as a new story was added 🤔

cheshire137 commented:
Hmm, checking the failure for the vrt-reports build, it shows a failure in an IconButton component:

<img width="998" height="319" alt="screenshot of VRT report in browser with IconButton failing" src="https://github.com/user-attachments/assets/4e06843b-043f-42f4-8c31-7bff059655ec" />

That seems unrelated to my changes.

cheshire137 commented:
Storybook preview: https://primer-8f3e67231c-13348165.drafts.github.io/storybook/?path=/story/components-toggleswitch-features--with-custom-labels

cheshire137 commented:
Ahh I didn't notice the bot had added IconButton snapshot changes when I added the 'update snapshots' label. I reverted those since I'm not modifying that component in this branch. Thanks to @llastflowers for the tip!
